File: app/src/main/java/com/example/guesstheword/HomeFragment.java
package com.example.guesstheword;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.navigation.Navigation;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;

public class HomeFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_home, container, false);

        Button buttonK = rootView.findViewById(R.id.K);
        buttonK.setOnClickListener(view -> navigateToAlphabetFragment("K"));

        return rootView;
    }

    private void navigateToAlphabetFragment(String buttonId) {
        HomeFragmentDirections.ActionHomeFragmentToAlphabetFragment action =
                HomeFragmentDirections.actionHomeFragmentToAlphabetFragment(buttonId);
        Navigation.findNavController(requireView()).navigate(action);
    }
}

File: app/src/main/java/com/example/guesstheword/HomeDir/HomeFragment.kt
package com.example.guesstheword.HomeDir

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.guesstheword.R


class HomeFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_home, container, false)

        // RecyclerView setup
        val recyclerView = view.findViewById<RecyclerView>(R.id.recyclerView)
        recyclerView.layoutManager = GridLayoutManager(context, 2)
        recyclerView.adapter = LetterAdapter { letter ->
            // Handle item click
            Toast.makeText(context, "Clicked: $letter", Toast.LENGTH_SHORT).show()
        }

        return view
    }
}
